stats.py

```diff
@@ -297,6 +299,74 @@ def basename():
+      patch = lambda s: s.replace(s.split()[0], '')
+      if len(lines) < 3: return None  # need 3 insts at least
+      mov_line = patch(lines[0])
+      dest_reg = x86.get('dst', mov_line)
+      dest_subs = x86.sub_regs(dest_reg)
+      # dest reg in 2nd line -> no candidate
+      # a. if dest reg is dest in 2nd line and fusion occurs -> not candidate
+      # b. if dest reg is dest in 2nd line and no fusion ->
+      # no candidate and disables next OPs to be candidates because dest reg got modified
+      # c. if dest reg is src in 2nd line -> dest reg value is used before OP, cancels candidate
+      if C.any_in(dest_subs, lines[1]): return None
+      to_check = lines[2:]
+      for i, line in enumerate(to_check):
+        line = patch(line)
+        if x86.get('dst', line) == dest_reg:  # same dest reg
+          # jcc macro-fusion disables candidate
+          if i < len(to_check) - 1 and x86.is_jcc_fusion(line, patch(to_check[i+1])): return None
+          ld_fusion, mov_fusion = x86.is_ld_op_fusion(mov_line, line), x86.is_mov_op_fusion(mov_line, line)
+          if not ld_fusion and not mov_fusion: return None
+          # check if dest reg was used as src before OP or OP src reg was ever modified
+          srcs = x86.get('srcs', line)
+          assert len(srcs) == 1
+          src_reg = srcs[0]
+          for x in range(1, i + 2):
+            if re.search(x86.CMOV, x86.get('inst', lines[x])): return None  # CMOV will use wrongly modified RFLAGS
+            if x86.is_sub_reg(x86.get('dst', lines[x]), src_reg): return None  # OP src reg was modified before OP
+            if C.any_in(dest_subs, lines[x]): return None  # dest reg used before OP
+          # candidate found
+          new_hotness = int(hotness(lines[0]))
+          if ld_fusion: stats_data['LD-OP'] += new_hotness
+          else: stats_data['MOV-OP'] += new_hotness
+          # append candidate block to log
+          header, tail = lines[0][:25] + "\n", lines[i+2][:25] + "z\n"  # headers to differentiate blocks
+          block_list = [header] + lines[0:i+3] + [tail]
+          C.fappend(''.join(block_list), cands_log, end='')
+      return None
+    if os.path.exists(cands_log): os.remove(cands_log)
+    # for each hotness block, create a list of the lines then check
+    with open(hitcounts, "r") as hits:
+      for line in hits:
+        # check blocks starting with not store MOV
+        if not condition(line) and not block: continue
+        if not block:  # new block first line found
+          block = [line]
+          hotness_key = hotness(line)
+          continue
+        # append lines from the same basic block (by hotness)
+        if hotness(line) == hotness_key: block.append(line)
+        else:  # basic block end, check candidates
+          for i, block_line in enumerate(block):
+            if condition(block_line): find_cand(block[i:])
+          if condition(line):
+            hotness_key, block = hotness(line), [line]  # restart for next block
```

Comment:
how about making this a tiny restart() local function that is invoked here, and in lines 354-355, and also for next line when overriding default args, for example.